From ed046d4b6be0d778ff18ceba768bc40eada3c025 Mon Sep 17 00:00:00 2001
From: Alexey Gladkov <gladkov.alexey@gmail.com>
Date: Thu, 24 Feb 2011 14:43:44 +0300
Subject: [BUILTIN] Fix corruption of reads with byte 0x81
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Starting with commit 55c46b dash removes CTLESC bytes ('\x81')
from read sequence. This leads to breakage of some UTF8
characters. Like in commit f8231a, this change fixes corruption
by removing the faulty code.

Testcase:

$ /usr/bin/printf '[\u0441]\n'
[с]
$ /usr/bin/printf '[\u0441]\n' |dash -c 'read c; printf "%s\n" "$c"'
[Ñ]

Signed-off-by: Alexey Gladkov <gladkov.alexey@gmail.com>
Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 src/miscbltin.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/src/miscbltin.c b/src/miscbltin.c
index 653c92f..4e88e8d 100644
--- a/src/miscbltin.c
+++ b/src/miscbltin.c
@@ -112,14 +112,12 @@ readcmd_handle_line(char *line, char **ap, size_t len)
 			 * will not modify the length of the string */
 			offset = sl->text - s;
 			remainder = backup + offset;
-			rmescapes(remainder);
 			setvar(*ap, remainder, 0);
 
 			return;
 		}
 		
 		/* set variable to field */
-		rmescapes(sl->text);
 		setvar(*ap, sl->text, 0);
 		sl = sl->next;
 	} while (*++ap);
-- 
1.7.4.1

